src/Swift.Bindings/src/Emitter/StringCSharpEmitter/Handler/TypeHandler.cs

```diff
@@ -0,0 +1,189 @@
+            {
+                unsafe
+                {
+                    // Apply struct layout attributes
+                    writer.WriteLine($"[StructLayout(LayoutKind.Sequential, Size = {swiftTypeInfo.Value.ValueWitnessTable->Size})]");
+                }
+            }
+            writer.WriteLine($"public unsafe struct {structDecl.Name} {{");
+            writer.Indent++;
+
+            // Emit each field in the struct
+            foreach (var fieldDecl in structDecl.Fields)
+            {
+                string accessModifier = fieldDecl.Visibility == Visibility.Public ? "public" : "private";
+                writer.WriteLine($"{accessModifier} {fieldDecl.TypeIdentifier.Name} {fieldDecl.Name};");
+
+                // TODO: Fix memory access violation
+                // // Verify field against Swift type information
+                // if (swiftTypeInfo.HasValue && !VerifyFieldRecord(swiftTypeInfo.Value, structDecl.Fields.IndexOf(fieldDecl), fieldDecl))
+                // {
+                //     Console.WriteLine("Field record does not match the field declaration");
+                // }
+            }
+            writer.WriteLine();
+
+            foreach (BaseDecl baseDecl in structDecl.Declarations)
+                base.HandleBaseDecl(writer, baseDecl, conductor, typeDatabase);
+            writer.Indent--;
+            writer.WriteLine("}");
+        }
+
+        /// <summary>
+        /// Verify field record with the Swift type information.
+        /// </summary>
+        private unsafe bool VerifyFieldRecord(SwiftTypeInfo swiftTypeInfo, int fieldIndex, FieldDecl fieldDecl)
+        {
+            // Access the field descriptor using pointer arithmetic
+            FieldDescriptor* desc = (FieldDescriptor*)IntPtr.Add(
+                (IntPtr)(((StructDescriptor*)swiftTypeInfo.Metadata->TypeDescriptor))->NominalType.FieldsPtr.Target,
+                IntPtr.Size * fieldIndex
+            );
+
+            // Ensure the field number is within bounds
+            if (desc->NumFields <= fieldIndex)
+            {
+                return false;
+            }
+
+            FieldRecord* fieldRecord = desc->GetFieldRecord(fieldIndex);
+
+            // Check field name
+            if ((System.Runtime.InteropServices.Marshal.PtrToStringAnsi((IntPtr)fieldRecord->Name.Target) ?? string.Empty) != fieldDecl.Name)
+            {
+                return false;
+            }
+
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Factory class for creating instances of ClassHandler.
+    /// </summary>
+    public class ClassHandlerFactory : IFactory<BaseDecl, ITypeHandler>
+    {
+        /// <summary>
+        /// Determines if the factory handles the specified declaration.
+        /// </summary>
+        /// <param name="decl">The base declaration.</param>
+        public bool Handles(BaseDecl decl)
+        {
+            return decl is TypeDecl && decl is ClassDecl;
```

Comment:
`decl is TypeDecl &&` isn't needed. If the type is a `ClassDecl` it is always a `TypeDecl`